matchzoo/preprocessors/units/tokenize.py

```diff
@@ -15,3 +17,104 @@ def transform(self, input_: str) -> list:
         :return tokens: tokenized tokens as a list.
         """
         return nltk.word_tokenize(input_)
+
+
+class ChineseTokenize(Unit):
+    """Process unit for text containing Chinese tokens."""
+
+    def transform(self, input_: str) -> str:
+        """
+        Process input data from raw terms to processed text.
+
+        :param input_: raw textual input.
+
+        :return output: text with a blank between adjacent Chinese tokens.
+        """
+        output = []
+        for char in input_:
+            cp = ord(char)
+            if is_chinese_char(cp):
+                output.append(" ")
+                output.append(char)
+                output.append(" ")
+            else:
+                output.append(char)
+        return "".join(output)
+
+
+class BasicTokenize(Unit):
+    """Process unit for text tokenization."""
+
+    def transform(self, input_: str) -> list:
+        """
+        Process input data from raw terms to list of tokens.
+
+        :param input_: raw textual input.
+
+        :return tokens: tokenized tokens as a list.
+        """
+        orig_tokens = whitespace_tokenize(input_)
+        split_tokens = []
+        for token in orig_tokens:
+            split_tokens.extend(run_split_on_punc(token))
+
+        output_tokens = whitespace_tokenize(" ".join(split_tokens))
+        return output_tokens
+
+
+class WordPieceTokenize(Unit):
+    """Process unit for text tokenization."""
+
+    def __init__(self, vocab: dict, max_input_chars_per_word: int = 200):
+        """Initialization."""
+        self.vocab = vocab
+        self.unk_token = '[UNK]'
+        self.max_input_chars_per_word = max_input_chars_per_word
+
+    def transform(self, input_: list) -> list:
+        """
+        Tokenizes a piece of text into its word pieces.
+
+        This uses a greedy longest-match-first algorithm to perform
+         tokenization using the given vocabulary.
+
+        For example:
+          input = "unaffable"
+          output = ["un", "##aff", "##able"]
+
+        :param input_: token list.
+
+        :return tokens: A list of wordpiece tokens.
+        """
+        output_tokens = []
+        for token in input_:
+            chars = list(token)
+            if len(chars) > self.max_input_chars_per_word:
+                output_tokens.append(self.unk_token)
+                continue
+
+            is_bad = False
```

Comment:
can you rename this variable, difficult to understand.

Reply:
rename it as `failed_match`?

Reply:
@jellying the logic of this function is a bit complicate, I need to read it through to better understand it.. So I don't know what you wanna do here.

Reply:
For example, we need to tokenize the word 'standard' and vocab = ['stand', 'dard'], we first match the prefix 'stand', but residual text  'ard' is not existed in vocab and 'a', 'r', 'd' neither. Without proper longest pieces in vocab to construct the target word, so the `is_bad` set `True` and we mark it as '[unk]'.

Reply:
@jellying then it should be something like `unknow_suffix`